Automattic/woocommerce-payments
# Updates to the Inbox Note logic.

Fixes #6441 

#### Changes proposed in this Pull Request





- Tidy up the inbox note display logic to make sure Stripe account is connected, that the account is not partially onboarded, and to make sure progressive onboarding isn't in progress.





#### Testing instructions



* Remove your connected account (you can do this by editing the `account_type` in the `wcpay_accounts` table on `localhost:8087` to something other than `wcpay`).
* Run `npm run listen` on the server.
* Comment out [this check](https://github.com/Automattic/woocommerce-payments/blob/5630fa6298134787b4c75d9fe8ca5015bbca0965/includes/notes/class-wc-payments-notes-additional-payment-methods.php#L42) temporarily - there are lots of ways of enabling UPE via various different flags so we'll just comment it out temporarily for simplicity. Alternatively, you can turn off UPE by going to `Overview > Settings` , clicking the `...`next to `Payment Methods` and then clicking `disable`.
* Onboard a new progressive onboarding account. When it gets to the point where payments are accepted but deposits are not yet enabled, stop and check the `Overview` page. You should NOT see the inbox note pictured below.
* Now, complete the full onboarding to become fully enabled. After you finish, you should now see the inbox note. If you properly disabled UPE, you should be able to click it to be redirected to the APM setup page.

<img width="724" alt="Screenshot 2023-09-07 at 14 33 35" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/940f82eb-5a11-47f7-8a26-b613aba261c2">



-------------------


**Post merge**
